### PR TITLE
Add menu option for `external_url` if present

### PR DESCRIPTION
**Pull request by Half-Shot**

### CODE REVIEW

File: src/components/views/context_menus/MessageContextMenu.js [hunk 1]
```diff
@@ -74,7 +74,7 @@ module.exports = React.createClass({
         if (this.props.onFinished) this.props.onFinished();
     },
 
-    onPermalinkClick: function() {
+    onLinkClick: function() {
```
Comment (dbkr) [inline src/components/views/context_menus/MessageContextMenu.js]: If you're re-using this, I'd call it 'closeMenu'

File: src/components/views/context_menus/MessageContextMenu.js [hunk 2]
```diff
@@ -156,6 +157,17 @@ module.exports = React.createClass({
             </div>
         );
 
+        // Bridges can provide a 'external_url' to link back to the source.
+        if( typeof(this.props.mxEvent.event.content.external_url) === "string") {
+          externalURLButton = (
+              <div className="mx_MessageContextMenu_field">
+                  <a target="_blank" href={ this.props.mxEvent.event.content.external_url }
+                     onClick={ this.onLinkClick }>Source URL</a>
```
Comment (dbkr) [inline src/components/views/context_menus/MessageContextMenu.js]: rel="noopener" please: https://mathiasbynens.github.io/rel-noopener/

File: src/components/views/context_menus/MessageContextMenu.js [hunk 3]
```diff
@@ -146,7 +147,7 @@ module.exports = React.createClass({
         permalinkButton = (
             <div className="mx_MessageContextMenu_field">
                 <a href={ "https://matrix.to/#/" + this.props.mxEvent.getRoomId() +"/"+ this.props.mxEvent.getId() }
-                   target="_blank" onClick={ this.onPermalinkClick }>Permalink</a>
+                   target="_blank" onClick={ this.onLinkClick }>Permalink</a>
```
Comment (dbkr) [inline src/components/views/context_menus/MessageContextMenu.js]: We should also add rel="noopener" here, which we seem to have forgotten (although presumably we sort of trust matrix.to)

Comment (Half-Shot) [inline src/components/views/context_menus/MessageContextMenu.js]: Didn't realise that was a thing. Neat.

File: src/components/views/context_menus/MessageContextMenu.js (hunk 2 above)
Comment (Half-Shot) [inline src/components/views/context_menus/MessageContextMenu.js]: I'll add it to both, just because it makes me feel better.

**Pull request merged**